src/test/java/GitDifferTest.java

```diff
@@ -0,0 +1,88 @@
+import org.variantsync.diffdetective.variation.diff.DiffTree;
+import org.variantsync.diffdetective.variation.diff.parse.DiffTreeParseOptions;
+import org.variantsync.diffdetective.variation.diff.parse.DiffTreeParser;
+import org.variantsync.diffdetective.variation.diff.serialize.Format;
+import org.variantsync.diffdetective.variation.diff.serialize.LineGraphExporter;
+import org.variantsync.diffdetective.variation.diff.serialize.edgeformat.ChildOrderEdgeFormat;
+import org.variantsync.diffdetective.variation.diff.serialize.nodeformat.FullNodeFormat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class GitDifferTest {
+    private final static Path testDir = Constants.RESOURCE_DIR.resolve("diffs").resolve("differ");
+    private final static Path REPOS_DIR = Path.of("src").resolve("test").resolve("resources").resolve("repos");
+
+    private static final String[] commitIDs = new String[] {"a032346092e47048becb36a7cb183b4739547370", "4a4c1db1b192e221d8e25460d6d1128d1bdd0c0d", "d7eeba1b94f3d0c0ebed3457eea8fa8537143348"};
+
+    public static Stream<String> testCommits() throws IOException {
+        return Arrays.stream(commitIDs);
+    }
+
+    @ParameterizedTest
+    @MethodSource("testCommits")
+    public void test(String basename) throws IOException, DiffParseException {
+        testCase(basename);
+    }
+
+    private static Repository repo() {
+        final Path marlinPath = Path.of(".")
+                .resolve(REPOS_DIR)
+                .resolve("test-spl.zip");
+        return Repository
+                .fromZip(marlinPath, "test-spl")
+                .setDiffFilter(DiffFilter.ALLOW_ALL)
+                .setParseOptions(new PatchDiffParseOptions(PatchDiffParseOptions.DiffStoragePolicy.REMEMBER_FULL_DIFF, new DiffTreeParseOptions(
+                        false,
+                        false
+                )));
+    }
+
+    public void testCase(String commitHash) throws IOException {
+        Repository repo = repo();
+        final CommitDiff commitDiff = DiffTreeParser.parseCommit(repo, commitHash);
+
+        for (final PatchDiff patch : commitDiff.getPatchDiffs()) {
+            Assertions.assertTrue(patch.isValid());
+            var actualPath = testDir.resolve(commitHash + "_actual.lg");
+            var expectedPath = testDir.resolve(commitHash + ".lg");
+
+            DiffTree diffTree = patch.getDiffTree();
+
+            try (var output = IO.newBufferedOutputStream(actualPath)) {
+                new LineGraphExporter(new Format(new FullNodeFormat(), new ChildOrderEdgeFormat()))
+                        .exportDiffTree(diffTree, output);
+            }
+
+            try (
+                    var expectedFile = Files.newBufferedReader(expectedPath);
+                    var actualFile = Files.newBufferedReader(actualPath);
```

Comment:
It seems as if you are writing the actual file to disk and then read it again. I guess it should be possible to keep it in memory and then pass the respective buffer to `IOUtils.contentEqualsIgnoreEOL`?

Reply:
You can leave as is for now. We can still refactor tests later.

Reply:
AlexanderSchultheiss's assumption is correct. That was the original intention. I will look into only writing the `*_actual*` files to disk if the test fails at some point.